platform/sun50i/devices.c

```diff
@@ -228,6 +251,13 @@ static struct device r_pio = {
 	.clocks  = CLOCK_PARENT(r_ccu, R_CCU_CLOCK_R_PIO),
 };
 
+static struct device ths = {
+	.name         = "ths",
+	.regs         = DEV_THS,
+	.drv          = &sun8i_thermal_driver.drv,
+	.subdev_count = 0,
```

Comment:
This should not be here. But you do need the `.clocks`

Reply:
That value is set in sun8i-thermal.c's function. '.clocks' was added

Reply:
right. since it's set there, it doesn't need to be set here (and even if it wasn't set there, zero is the default, so setting it to zero is redundant).

Reply:
It still should not be set here, because 1) it's overwritten by the probe function, and 2) zero is the default

Reply:
done